src/shapes/textbox.class.js

```diff
@@ -328,14 +337,21 @@
         words.push([]);
       }
       desiredWidth -= reservedSpace;
-      for (var i = 0; i < words.length; i++) {
+      var data = words.map(function (word) {
         // if using splitByGrapheme words are already in graphemes.
-        word = splitByGrapheme ? words[i] : fabric.util.string.graphemeSplit(words[i]);
+        word = splitByGrapheme ? word : this.graphemeSplit(word);
+        var width = this._measureWord(word, lineIndex);
+        largestWordWidth = Math.max(width, largestWordWidth);
```

Comment:
what is going on here? We are measuring twice, without using the offset, tha means without counting styles, and that means that the measuring can be extremely wrong.
You can access offset in that loop, but why we need to loop over words twice?